storage/heap/hp_extra.c

```diff
@@ -34,6 +34,7 @@ int heap_extra(register HP_INFO *info, enum ha_extra_function function)
   switch (function) {
   case HA_EXTRA_RESET_STATE:
     heap_reset(info);
+	// fallthrough
```

Comment:
Isn't it an actual "break" missing here? Also formatting

Reply:
Ops, you are right. I wasn't sure about it, as setting the flags during the reset also makes sense, but I checked now and in 8.0, there is a break here.